packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -309,3 +310,49 @@ async function tryToFixInvalidFuncsWithValidIndexAlternative({
 		}
 	}
 }
+
+/**
+ * Tries to fix invalid dynamic ISR functions that have valid prerendered children.
+ *
+ * The Vercel CLI might not generated a prerender config for a dynamic ISR function, depending
+ * on the Vercel CLI version. Therefore, we also check if valid prerendered routes were created
+ * for the dynamic route to determine if the function can be ignored.
+ *
+ * @param collectedFunctions Collected functions from the Vercel build output.
+ */
+async function tryToFixInvalidDynamicISRFuncs({
+	prerenderedFunctions,
+	invalidFunctions,
+	ignoredFunctions,
+}: CollectedFunctions) {
+	if (invalidFunctions.size === 0) {
+		return;
+	}
+
+	const prerenderedFunctionEntries = [...prerenderedFunctions.values()];
+
+	for (const [fullPath, fnInfo] of invalidFunctions.entries()) {
+		const fnPathWithoutRscOrFuncExt = fnInfo.relativePath.replace(
+			/(\.rsc)?\.func$/,
+			'',
+		);
+
+		const isDynamicISRFunc =
+			fnInfo.config.operationType === 'ISR' &&
+			/\/\[[\w-]+\]$/.test(fnPathWithoutRscOrFuncExt);
+
+		if (isDynamicISRFunc) {
+			const matchingPrerenderedChildFunc = prerenderedFunctionEntries.find(
+				fnInfo => fnInfo.sourcePath === fnPathWithoutRscOrFuncExt,
+			);
+
+			if (matchingPrerenderedChildFunc) {
+				ignoredFunctions.set(fullPath, {
+					reason: 'invalid dynamic isr route with valid prerendered children',
+					...fnInfo,
+				});
+				invalidFunctions.delete(fullPath);
+			}
```

Comment:
Is this correct? shouldn't this check if all children are valid?

```suggestion
			const childrenArePrerendered = prerenderedFunctionEntries.every(
				fnInfo => fnInfo.sourcePath === fnPathWithoutRscOrFuncExt,
			);

			if (childrenArePrerendered) {
				ignoredFunctions.set(fullPath, {
					reason: 'invalid dynamic isr route with valid prerendered children',
					...fnInfo,
				});
				invalidFunctions.delete(fullPath);
			}
```

Reply:
it only need to know if there was a prerendered child relating to the dynamic route to figure out if anything was prerendered or if it's an invalid nodejs function.

You can have non-dynamic routes at the same layer as dynamic routes, so it should only check until it finds one

Reply:
> it only need to know if there was a prerendered child relating to the dynamic route to figure out if anything was prerendered or if it's an invalid nodejs function.
> 
> You can have non-dynamic routes at the same layer as dynamic routes, so it should only check until it finds one

ah ok I see, thanks! 🙂👍